code/modules/clothing/shoes/colour.dm

```diff
@@ -11,202 +12,217 @@
 
 /obj/item/clothing/shoes/science
 	name = "scientist shoes"
 	desc = "A pair of treated purple and white shoes resistant to chemical and gas spills."
 	icon_state = "scientist"
+	item_state = "purple"
 	gas_transfer_coefficient = 0.01
 	permeability_coefficient = 0.01
 
 /obj/item/clothing/shoes/chemist
 	name = "pharmacist shoes"
 	desc = "A pair of orange and white shoes resistant to biological and chemical hazards."
 	icon_state = "chemist"
+	item_state = "orange"
 	permeability_coefficient = 0.01
 	armor = list(melee = 0, bullet = 0, laser = 0,energy = 0, bomb = 0, bio = 90, rad = 0)
 
 /obj/item/clothing/shoes/biochem
 	name = "biochemist shoes"
 	desc = "A pair of red and white shoes resistant to biological and chemical hazards."
 	icon_state = "biochem"
+	item_state = "red"
 	permeability_coefficient = 0.01
 	armor = list(melee = 0, bullet = 0, laser = 0,energy = 0, bomb = 0, bio = 90, rad = 0)
 
 /obj/item/clothing/shoes/psych
 	name = "psychologist shoes"
 	desc = "A pair of teal and white shoes resistant to biological and chemical hazards."
 	icon_state = "psych"
+	item_state = "blue"
 	permeability_coefficient = 0.01
 	armor = list(melee = 0, bullet = 0, laser = 0,energy = 0, bomb = 0, bio = 90, rad = 0)
 
 /obj/item/clothing/shoes/surgeon
 	name = "surgeon shoes"
 	desc = "A pair of light blue and white shoes resistant to biological and chemical hazards."
 	icon_state = "surgeon"
+	item_state = "blue"
 	permeability_coefficient = 0.01
 	armor = list(melee = 0, bullet = 0, laser = 0,energy = 0, bomb = 0, bio = 90, rad = 0)
 
-/obj/item/clothing/shoes/rainbow
-	name = "rainbow shoes"
-	desc = "A pair of overly colorful shoes."
-	icon_state = "rain_bow"
-
-/obj/item/clothing/shoes/orange
-	name = "orange shoes"
-	desc = "A pair of easily noticed, reflective orange shoes."
-	icon_state = "orange"
-	var/obj/item/handcuffs/chained = null
-
-/obj/item/clothing/shoes/orange/proc/attach_cuffs(var/obj/item/handcuffs/cuffs, mob/user)
-	if (src.chained) return
-
-	user.drop_from_inventory(cuffs,src)
-	src.chained = cuffs
-	src.slowdown = 15
-	src.icon_state = "orange1"
-
-/obj/item/clothing/shoes/orange/proc/remove_cuffs(mob/user as mob)
-	if (!src.chained) return
-
-	user.put_in_hands(src.chained)
-	src.chained.add_fingerprint(user)
-
-	src.slowdown = initial(slowdown)
-	src.icon_state = "orange"
-	src.chained = null
-
-/obj/item/clothing/shoes/orange/attack_self(mob/user as mob)
-	..()
-	remove_cuffs(user)
-
-/obj/item/clothing/shoes/orange/attackby(H as obj, mob/user as mob)
-	..()
-	if (istype(H, /obj/item/handcuffs))
-		attach_cuffs(H, user)
-
 /obj/item/clothing/shoes/flats
 	desc = "A pair of black, low-heeled women's flats."
 	name = "black dress flats"
 	icon = 'icons/obj/clothing/cheongsams.dmi'
-	icon_state = "dressflatsblack"
-	item_state = "dressflatsblack"
+	icon_state = "blackdf"
+	item_state = "blackdf"
 	species_restricted = null
 	contained_sprite = 1

```

Comment:
As the flats are a contained sprite, the inhand sprites can't apply as it looks for the sprites in 'icons/obj/clothing/cheongsams.dmi'.